source/release-notes/1.6.txt

```diff
@@ -0,0 +1,145 @@
+use of exclusion filters through the ``excludeNamespaces`` parameter.
+
+For more information, see :ref:`c2c-filtered-sync`.
+
+Regular Expression Filters
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Both inclusion and exclusion filters in :ref:`c2c-filtered-sync` now 
+support matching databases and collections using Regular Expressions 
+with the ``databaseRegex`` and ``collectionRegex`` fields.
+
+For more information, see :ref:`c2c-filter-regex`.
+
+7.0 Support
+~~~~~~~~~~~
+
+Starting in 1.6.0, ``mongosync`` supports sync with MongoDB 7.0.
+
+Other Notes
+~~~~~~~~~~~
+
+New Features:
+
+- Sped up the commit process by parallelizing metadata cleanup. 
+
+- Reduced memory allocation during change event application by approximately 
+  30% for pre-6.0 releases of MongoDB Server versions.
+
+- Changed telemetry to also send error messages to Segment when ``mongosync``
+  exits with a fatal error.
+
+- Added support for ``mongosync`` to sync between source and destination 
+  clusters that have the same major and minor versions, but different 
+  patch versions.
+
+- Upgraded build to use Go 1.20.
+
+- ``mongosync`` now periodically logs when server operations take a longer 
+  than expected time.
+
+Issues Fixed:
+
+- Fixed a bug introduced in 1.5, where ``mongosync`` could fail to apply change
+  events for a document if they occur shortly after the document is copied
+  during the Collection Copy phase.
+
+- Fixed a bug where ``mongosync`` could crash due to the source cluster
+  containing legacy indexes with unknown index options.
+
+- Fixed a bug where ``mongosync`` could crash with an error during change event
+  application, if the application fails to read 500 documents within 5 minutes
+  or reaches the end of the oplog when reading from change streams on the
+  source cluster.
+
+- Fixed a bug where ``mongosync`` could crash when indexes apply constraints to
+  collection documents.
+
+  This crash occured when such an index was added or dropped at nearly the same
+  time as an index that violated these constrains was deleted or inserted.  It
+  would manifest with an error like:
+
+  .. code-block:: text
+     :copyable: false
+
+     Ambiguous field name found in array (do not use numeric field names in embedded elements in an array)
+
+  or a similar error. This fix makes this crash less likely, but it does not
+  elimiate the possibility that it will occur.
+
+- Fixed a bug where ``mongosync`` can fail to copy certain documents if the
+  insertion of documents causes duplicate key errors at the same time as a
+  write concern error. This bug has been present in mongosync since its first
+  release and may have arisen if the destination cluster logs reported a write
+  concern error alongside a duplicate key error.
+
+- Fixed a bug introduced in ``mongosync`` 1.5 where change events may be
+  skipped during the Collection Copy phase when there is little write activity
+  on the source cluster.
```

Comment:
The team would like to change this bullet to:

> Fixed a rare bug introduced in Mongosync v1.5.0 and discovered by our internal testing where there may be a small continuity gap between Oplog Rollover Resilience (ORR) cycles in a specific situation, causing the ORR mechanism to potentially miss a change event to be applied during the Collection Copy phase. This is an extreme case as it requires consistently little write activity on the source cluster (i.e. Mongosync’s replication lag is consistently near zero during Collection Copy) as well as the change event to fall in specific `_id` ranges between ORR cycles.

Reply:
Fixed